tests/integration/bundle/test_bundle.py

```diff
@@ -0,0 +1,166 @@
+    # relation
+    await scale_application(ops_test, PGB, 3)
+
+
+@pytest.mark.bundle
+async def test_kill_pg_primary(ops_test: OpsTest):
+    """Kill primary, check that all proxy instances switched traffic for a new primary."""
+    # Get postgres primary through action
+    unit_name = ops_test.model.applications[PG].units[0].name
+    action = await ops_test.model.units.get(unit_name).run_action("get-primary")
+    action = await action.wait()
+    primary = action.results["primary"]
+
+    # Get primary connection string from each pgbouncer unit and assert they're pointing to the
+    # correct PG primary
+    old_primary_ip = ops_test.model.units.get(primary).public_address
+    for unit in ops_test.model.applications[PGB].units:
+        unit_cfg = await get_cfg(ops_test, unit.name)
+        assert unit_cfg["databases"]["mailman3"]["host"] == old_primary_ip
+        assert unit_cfg["databases"]["mailman3_standby"]["host"] != old_primary_ip
+
+    await ops_test.model.destroy_units(primary)
+    await ops_test.model.wait_for_idle(
+        apps=[PG, PGB, MAILMAN3_CORE_APP_NAME], status="active", timeout=600
+    )
+
+    # Do some CRUD operations using Mailman3 Core client to ensure it's still working.
+    mailman_unit = ops_test.model.applications[MAILMAN3_CORE_APP_NAME].units[0]
+    action = await mailman_unit.run("mailman info")
+    result = action.results.get("Stdout", action.results.get("Stderr", None))
+    domain_name = "canonical.com"
+    credentials = (
+        result.split("credentials: ")[1].strip().split(":")
+    )  # This outputs a list containing username and password.
+    client = Client(
+        f"http://{mailman_unit.public_address}:8001/3.1", credentials[0], credentials[1]
+    )
+
+    # Create a domain and list the domains to check that the new one is there.
+    client.create_domain(domain_name)
+    assert domain_name in [domain.mail_host for domain in client.domains]
+
+    # Assert pgbouncer config points to the new correct primary
+    unit_name = ops_test.model.applications[PG].units[0].name
+    action = await ops_test.model.units.get(unit_name).run_action("get-primary")
+    action = await action.wait()
+    new_primary = action.results["primary"]
+    new_primary_ip = ops_test.model.units.get(new_primary).public_address
+    assert new_primary_ip != old_primary_ip
+    for unit in ops_test.model.applications[PGB].units:
+        unit_cfg = await get_cfg(ops_test, unit.name)
+        assert unit_cfg["databases"]["mailman3"]["host"] == new_primary_ip
+        assert unit_cfg["databases"]["mailman3_standby"]["host"] != new_primary_ip
+        assert unit_cfg["databases"]["mailman3_standby"]["host"] != old_primary_ip
+
+
+@pytest.mark.bundle
+async def test_discover_dbs(ops_test: OpsTest):
+    """Check that proxy discovers new members when scaling up postgres charm."""
+    scale_application(ops_test, PG, 3)
```

Comment:
It seems that, for some reason, the PostgreSQL application isn't being scaled correctly here.